File: Movie/API/CreditsService.swift
//
//  CreditsService.swift
//  Movie
//
//

import Foundation
import Alamofire

class CreditsService {
    
    func getCredits(id: Int) async throws -> CreditsResponse {
        return try await AF.request(
            "\(ApiConstants.baseUrl)/movie/\(id)/credits?api_key=\(ApiConstants.apiKey)"
        )
        .serializingDecodable(CreditsResponse.self)
        .value
    }
}

File: shared/src/commonMain/kotlin/com/mehmetaliocak/movie_app/api/CreditsService.kt
package com.mehmetaliocak.movie_app.api

import com.mehmetaliocak.movie_app.models.CreditsResponse
import com.rickclephas.kmp.nativecoroutines.NativeCoroutines
import io.ktor.client.HttpClient
import io.ktor.client.call.body
import io.ktor.client.request.get
import org.koin.core.component.KoinComponent
import org.koin.core.component.inject

class CreditsService : KoinComponent {
    private val client: HttpClient by inject()

    @NativeCoroutines
    suspend fun getCredits(id: Long): CreditsResponse {
        return client.get("movie/$id/credits").body()
    }
}